packages/react/menu/src/Menu.tsx

```diff
@@ -621,6 +636,206 @@ const MenuItemIndicator = React.forwardRef((props, forwardedRef) => {
+          event.preventDefault();
+        }
+      })}
+    />
+  );
+}) as MenuSubMenuContentPrimitive;
+
+MenuSubMenuContent.displayName = SUB_MENU_CONTENT_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * MenuSubMenuTrigger
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_MENU_TRIGGER_NAME = 'MenuSubMenuTrigger';
+
+type MenuSubMenuTriggerOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof MenuItem>,
+  Polymorphic.OwnProps<typeof MenuAnchor>
+>;
+type MenuSubMenuTriggerPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuItem>,
+  MenuSubMenuTriggerOwnProps
+>;
+
+const MenuSubMenuTrigger = React.forwardRef((props, forwardedRef) => {
+  const { disabled, ...triggerProps } = props;
+  const context = useMenuContext(SUB_MENU_TRIGGER_NAME);
+  const subMenuContext = useSubMenuContext(SUB_MENU_TRIGGER_NAME);
+  const [mouseInteracting, setMouseInteracting] = React.useState(false);
+
+  return (
+    <MenuAnchor as={Slot}>
+      <MenuItem
+        aria-haspopup="menu"
+        aria-expanded={context.open ? true : undefined}
+        aria-controls={context.open ? subMenuContext.contentId : undefined}
+        data-state={getOpenState(context.open)}
+        disabled={disabled}
+        {...triggerProps}
+        ref={composeRefs(forwardedRef, subMenuContext.triggerRef)}
+        onSelect={composeEventHandlers(triggerProps.onSelect, (event) => {
+          event.preventDefault();
+          event.stopPropagation();
+        })}
+        onMouseMove={composeEventHandlers(triggerProps.onMouseMove, (event) => {
+          // Prevent refocusing trigger which causes premature menu close
+          event.preventDefault();
+
+          if (!disabled && !mouseInteracting && !context.open) {
+            setMouseInteracting(true);
+            event.currentTarget.focus();
+          }
+        })}
+        onMouseLeave={composeEventHandlers(triggerProps.onMouseLeave, (event) => {
+          // Prevent focusing content which causes premature menu close
+          event.preventDefault();
+          setMouseInteracting(false);
+        })}
+        /**
+         * We hook into focus to control menu visibility rather than mouse events
+         * This solves a sticky open state problem when certain browser controls (e.g. devtools inspect overlay) are given focus
+         * This is because mouse events continue to fire while `onDismiss` in `DismissableLayer` won't get called due to it being a focus outside check
+         */
+        onFocus={composeEventHandlers(triggerProps.onFocus, (event) => {
+          if (mouseInteracting) subMenuContext.onMouseOpen();
+        })}
+        onKeyDown={composeEventHandlers(triggerProps.onKeyDown, (event) => {
+          setMouseInteracting(false);
+          if (['Enter', ' ', 'ArrowRight'].includes(event.key)) subMenuContext.onKeyboardOpen();
```

Comment:
Should we provide `rtl` support for flipping the `ArrowRight / ArrowLeft` bindings and `side` prop default?